http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamRetry.scala

```diff
@@ -0,0 +1,67 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao.sam
+
+import cats.effect.Async
+import cats.mtl.Ask
+import org.broadinstitute.dsde.workbench.RetryConfig
+import org.broadinstitute.dsde.workbench.client.sam.ApiException
+import org.apache.http.HttpStatus
+import org.broadinstitute.dsde.workbench.util2.addJitter
+
+import java.net.SocketTimeoutException
+import scala.concurrent.duration._
+import org.broadinstitute.dsde.workbench.google2.tracedRetryF
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.typelevel.log4cats.StructuredLogger
+
+object SamRetry {
+
+  /**
+   * Initial delay of 1 second, exponential retry up to 5 times.
+   */
+  private val defaultSamRetryConfig =
+    RetryConfig(addJitter(1 seconds, 1 seconds), _ * 2, 5, isRetryable)
+
+  /**
```

Comment:
Borrowed this logic from TCL `SamRetry`: https://github.com/DataBiosphere/terra-common-lib/blob/develop/src/main/java/bio/terra/common/sam/SamRetry.java#L58